Code/Framework/AzFramework/Platform/iOS/AzFramework/Input/Devices/Gamepad/InputDeviceGamepad_iOS.mm

```diff
@@ -112,50 +113,37 @@ bool GetPhysicalKeyOrButtonText(const InputChannelId& inputChannelId,
     };
 
     ////////////////////////////////////////////////////////////////////////////////////////////////
-    AZ::u32 InputDeviceGamepad::GetMaxSupportedGamepads()
-    {
-        return GCControllerPlayerIndex4 + 1;
-    }
-
-    ////////////////////////////////////////////////////////////////////////////////////////////////
-    InputDeviceGamepad::Implementation* InputDeviceGamepad::Implementation::Create(
-        InputDeviceGamepad& inputDevice)
-    {
-        return aznew InputDeviceGamepadApple(inputDevice);
-    }
-
-    ////////////////////////////////////////////////////////////////////////////////////////////////
-    InputDeviceGamepadApple::InputDeviceGamepadApple(InputDeviceGamepad& inputDevice)
+    InputDeviceGamepadiOS::InputDeviceGamepadiOS(InputDeviceGamepad& inputDevice)
         : InputDeviceGamepad::Implementation(inputDevice)
         , m_rawGamepadState(GetDigitalButtonIdByBitMaskMap())
         , m_controller(nullptr)
         , m_wasPausedHandlerCalled(false)
     {
-        AZ_Assert(inputDevice.GetInputDeviceId().GetIndex() < InputDeviceGamepad::GetMaxSupportedGamepads(),
-                  "Creating InputDeviceGamepadApple with index %d that is greater than the max supported by the game controller framework: %d",
-                  inputDevice.GetInputDeviceId().GetIndex(), InputDeviceGamepad::GetMaxSupportedGamepads());
+        AZ_Assert(inputDevice.GetInputDeviceId().GetIndex() <= GCControllerPlayerIndex4,
+                  "Creating InputDeviceGamepadiOS with index %d that is greater than the max supported by the game controller framework: %d",
+                  inputDevice.GetInputDeviceId().GetIndex(), (GCControllerPlayerIndex4+1));
```

Comment:
Nit: If you can't still use GetMaxSupportedGamepads() here, it would be nice if you put "GCControllerPlayerIndex4+1" into a constant somewhere that this code and GetMaxSupportedGamepads() could both use.